packages/expect/src/matchers.ts

```diff
@@ -134,16 +134,41 @@ const matchers: MatchersObject = {
     ensureNumbers(received, expected, matcherName, options);
 
     let pass = false;
-    let expectedDiff = 0;
-    let receivedDiff = 0;
-
-    if (received === Infinity && expected === Infinity) {
-      pass = true; // Infinity - Infinity is NaN
-    } else if (received === -Infinity && expected === -Infinity) {
-      pass = true; // -Infinity - -Infinity is NaN
+    let expectedDiff: number | bigint = 0;
+    let receivedDiff: number | bigint = 0;
+
+    //If both arent numbers, then one might be bigint
+    if (typeof received === 'number' && typeof expected === 'number') {
+      if (received === Infinity && expected === Infinity) {
+        pass = true; // Infinity - Infinity is NaN
+      } else if (received === -Infinity && expected === -Infinity) {
+        pass = true; // -Infinity - -Infinity is NaN
+      } else {
+        // Set default precision for numbers
+        precision = isNaN(precision) ? 2 : precision;
+        expectedDiff = Math.pow(10, -precision) / 2;
+        receivedDiff = Math.abs(expected - received);
+        pass = receivedDiff < expectedDiff;
+      }
     } else {
-      expectedDiff = Math.pow(10, -precision) / 2;
-      receivedDiff = Math.abs(expected - received);
+      // Set default precision for big integers
+      precision = isNaN(precision) ? -2 : precision;
+      if (precision >= 0) {
+        throw new Error(
+          matcherErrorMessage(
+            matcherHint(matcherName, undefined, undefined, options),
+            `${EXPECTED_COLOR(
+              'precision',
+            )} value must be a negative number for BigInts`,
```

Comment:
Let’s be specific to say: value must be a negative integer for BigInts

Can you believe that the matcher had not been checking the type of the `precision` arguments?

Reply:
haha, cant imagine someone passing in a decimal, but im sure its been done. 🤦‍♂